module/Calculator/SpeedCalculator.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ *  @file SpeedCalculator.cpp
+ *  @brief 目標速度に対応するPWM値を算出するクラス
+ *  @author CHIHAYATAKU
+ */
+#include "SpeedCalculator.h"
+
+SpeedCalculator::SpeedCalculator(double _targetSpeed)
+  : rightTargetSpeed(_targetSpeed),
+    leftTargetSpeed(_targetSpeed),
+    rightPid(K_P, K_I, K_D, _targetSpeed),
+    leftPid(K_P, K_I, K_D, _targetSpeed)
+{
+  rightMotorPwm = Controller::getRightMotorPwm();
+  leftMotorPwm = Controller::getLeftMotorPwm();
+  int rightAngle = measurer.getRightCount();
+  int leftAngle = measurer.getLeftCount();
+  prevRightMileage = Mileage::calculateWheelMileage(rightAngle);
+  prevLeftMileage = Mileage::calculateWheelMileage(leftAngle);
+  int currentTime = timer.now();
+  prevRightTime = currentTime;
+  prevLeftTime = currentTime;
+}
+
+// 回頭用（2024版では使わないかもしれないが一応残しておく）
+SpeedCalculator::SpeedCalculator(double _rightTargetSpeed, double _leftTargetSpeed)
+  : rightTargetSpeed(_rightTargetSpeed),
+    leftTargetSpeed(_leftTargetSpeed),
+    rightPid(R_K_P, R_K_I, R_K_D, _rightTargetSpeed),
+    leftPid(R_K_P, R_K_I, R_K_D, _leftTargetSpeed)
+{
+  rightMotorPwm = Controller::getRightMotorPwm();
+  leftMotorPwm = Controller::getLeftMotorPwm();
+  int rightAngle = measurer.getRightCount();
+  int leftAngle = measurer.getLeftCount();
+  prevRightMileage = Mileage::calculateWheelMileage(rightAngle);
+  prevLeftMileage = Mileage::calculateWheelMileage(leftAngle);
+  int currentTime = timer.now();
+  prevRightTime = currentTime;
+  prevLeftTime = currentTime;
+}
+
+double SpeedCalculator::calculaterightMotorPwmFromTargetSpeed()
```

Comment:
calculateとrightが混ざって見ずらいかな。
個人的には、Motor消して、calculateRightPwmFromTargetSpeed()でも良さそう。
```suggestion
double SpeedCalculator::calculaterightMotorPwmFromTargetSpeed()
```